purchase_product_weight/models/purchase.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    weight_total = fields.Float(string='Total weight')
+    product_weight = fields.Float(
+        related='product_id.weight',
+        string='Product Weight')
+
+    @api.onchange('product_id', 'product_weight', 'product_qty')
+    def _compute_total_weight(self):
```

Comment:
For onchange, the method should named "_onchange_xxx" ?

Reply:
right, done!